ccm_web/server/src/migrations/2021.06.08T13.39.45.create-canvas-token.ts

```diff
@@ -0,0 +1,30 @@
+import { DataTypes } from 'sequelize'
+import { Migration } from '../migrator'
+
+export const up: Migration = async ({ context: sequelize }) => {
```

Comment:
This is an interesting case I found out.
I tried to become user and  was able to authorize as him and generated the token. I feel this would be wrong. We might need to deal with this issue.

I looked up in the LTI substitution variable $Canvas.masqueradingUser.id, If launch is by user as him this variable won't be populated
i.e 
```
"https://purl.imsglobal.org/spec/lti/claim/custom": {
    "roles": "TeacherEnrollment,Account Admin",
    "login_id": "userA", // canvas_id= 56
    "course_id": 44444444,
    "become_user": "$Canvas.masqueradingUser.id",
    "is_root_account_admin": true
  },
```
But if the launch is via becoming an user this id get populated telling who the the masqueraded user is

```
"https://purl.imsglobal.org/spec/lti/claim/custom": {
    "roles": "TeacherEnrollment,Account Admin",
    "login_id": "userB", // Masqueraded user
    "course_id": 44444444,
    "become_user": 56, // this is the masquerading user 
    "is_root_account_admin": true
  },
```

By `become_user` variable we can identify this a masquerading launch and we can handle this case. There may be better ways to handle this but this is just 20 minutes of research

**Note: Nothing to do here about the above case**

Reply:
Yeah, so separate issue, I agree we shouldn't allow this in production. What about in test environments? Seems like maybe it's safer to disallow it across the board.

Reply:
I've created this issue to address the masquerading user case: https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/105